Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/repo/DefaultContactDiaryRepository.kt

```diff
@@ -0,0 +1,205 @@
+        Timber.d("Adding location visit $contactDiaryLocationVisit")
+        executeWhenIdNotDefault(contactDiaryLocationVisit.id) {
+            val contactDiaryLocationVisitEntity = contactDiaryLocationVisit.toContactDiaryLocationVisitEntity()
+            contactDiaryLocationVisitDao.insert(contactDiaryLocationVisitEntity)
+        }
+    }
+
+    override suspend fun deleteLocationVisit(contactDiaryLocationVisit: ContactDiaryLocationVisit) {
+        Timber.d("Deleting location visit $contactDiaryLocationVisit")
+        executeWhenIdNotDefault(contactDiaryLocationVisit.id) {
+            val contactDiaryLocationVisitEntity = contactDiaryLocationVisit.toContactDiaryLocationVisitEntity()
+            contactDiaryLocationVisitDao.delete(contactDiaryLocationVisitEntity)
+        }
+    }
+
+    override suspend fun deleteAllLocationVisits() {
+        Timber.d("Clearing contact diary location visit table")
+        contactDiaryLocationVisitDao.deleteAll()
+    }
+
+    // Person
+    override val people: Flow<List<ContactDiaryPerson>> = contactDiaryPersonDao
+        .allEntries()
+        .map { it.sortByNameAndIdASC() }
+
+    override suspend fun addPerson(contactDiaryPerson: ContactDiaryPerson) {
+        Timber.d("Adding person $contactDiaryPerson")
+        contactDiaryPersonDao.insert(contactDiaryPerson.toContactDiaryPersonEntity())
+    }
+
+    override suspend fun updatePerson(contactDiaryPerson: ContactDiaryPerson) {
+        Timber.d("Updating person $contactDiaryPerson")
+        val contactDiaryPersonEntity = contactDiaryPerson.toContactDiaryPersonEntity()
+        executeWhenIdNotDefault(contactDiaryPersonEntity.personId) {
+            contactDiaryPersonDao.update(contactDiaryPersonEntity)
+        }
+    }
+
+    override suspend fun deletePerson(contactDiaryPerson: ContactDiaryPerson) {
+        Timber.d("Deleting person $contactDiaryPerson")
+        val contactDiaryPersonEntity = contactDiaryPerson.toContactDiaryPersonEntity()
+        executeWhenIdNotDefault(contactDiaryPersonEntity.personId) {
+            contactDiaryPersonDao.delete(contactDiaryPersonEntity)
+        }
+    }
+
+    override suspend fun deletePeople(contactDiaryPeople: List<ContactDiaryPerson>) {
+        Timber.d("Deleting people $contactDiaryPeople")
+        val contactDiaryPersonEntities = contactDiaryPeople
+            .map {
+                val contactDiaryPersonEntity = it.toContactDiaryPersonEntity()
+                executeWhenIdNotDefault(contactDiaryPersonEntity.personId)
+                return@map contactDiaryPersonEntity
+            }
+        contactDiaryPersonDao.delete(contactDiaryPersonEntities)
+    }
+
+    override suspend fun deleteAllPeople() {
+        Timber.d("Clearing contact diary person table")
+        contactDiaryPersonDao.deleteAll()
+    }
+
+    // Person encounter
+    private suspend fun List<ContactDiaryPersonEncounterEntity>.toContactDiaryPersonEncounterSortedList():
```

Comment:
as mentioned above: why do you not join the data on the DB?

Reply:
adjusted